lr_face/data_providers.py

```diff
@@ -117,6 +117,69 @@ def enfsi_data(resolution, year) -> PairsWithIds:
     return PairsWithIds(pairs=X, is_same_source=y, pair_ids=ids)
 
 
+def lfw_data(resolution) -> PairsWithIds:
+    folder = os.path.join('resources', 'lfw')
+    df = pd.read_csv(os.path.join(folder, 'pairs.txt'), sep="\t", names=['col1', 'col2', 'col3', 'col4'])
```

Comment:
If I understand the `pairs.txt` file correctly (I checked out the README at http://vis-www.cs.umass.edu/lfw/README.txt), it's just a sampling of possible pairs to report performance on, right? In total there seem to be 6000 pairs defined in this file. Do we want to limit ourselves to this arbitrary sampling of pairs? Because in theory, we could create pairs on the fly ourselves by just matching any two images of the same person for a positive pair, or two images of different people for a negative pair. Or have these pairs been carefully selected based on difficulty or some other factor?